prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/ModelNexus.java

```diff
@@ -660,36 +648,26 @@ private EReference getServiceForModel(EClass model, String serviceName) {
         return feature == null ? null : (EReference) feature;
     }
 
-    public Stream<EAttribute> getResourcesForService(EClass svcClass) {
-        return svcClass.getEAllAttributes().stream().filter(a -> {
-            EAnnotation annotation = a.getEAnnotation("metadata");
-            if (annotation == null) {
-                // TODO do we have to special case admin, or should it have a metadata
-                // annotation
-                return sensinactPackage.getAdmin().equals(a.getEContainingClass());
-            } else {
-                return annotation.eContents().stream().anyMatch(ModelMetadata.class::isInstance);
-            }
-        });
+    public Stream<ETypedElement> getResourcesForService(EClass svcClass) {
+        List<ETypedElement> result = new ArrayList<>(svcClass.getEAllAttributes());
+        result.addAll(svcClass.getEOperations());
```

Comment:
Is there a special type of `EOperation` that we should filter for (like there is with`ResourceAttribute`)?

Reply:
I already planed a bit ahead, when I want to work more directly with EObjects using EClasses that don't inherit from our model.

Reply:
Ok - we should add a comment here saying that it may be too permissive and then we should use an issue to plan how we recognise resource attributes in these cases.

Reply:
I've created #134 for this.